src/CSF.SRDashboard.Client/Components/AddDocumentComponent.razor.cs

```diff
@@ -0,0 +1,162 @@
+            this.ValidationMessageStore = new ValidationMessageStore(this.EditContext);
+            this.Applicant = this.GatewayService.GetApplicantInfoByCdn(Cdn);
+            this.DocumentForm = new AddDocumentModel();
+            this.MultipleSelectTitle = "Select";
+        }
+
+        private void OnInputChange(EventArgs e)
+        {
+
+        }
+        private async void HandleValidSubmit()
+        {
+            
+            this.DocumentTypes = PopulateDocumentTypes(this.DocumentForm.DocumentTypeList);
+            var isValid = Validate();
+
+           
+
+            Console.WriteLine("Valid Submit");
+            if (isValid)
+            {
+                this.Language = this.DocumentForm.Languages[this.DocumentForm.SelectValue - 1].Text;
+                var result = await DocumentServe.InsertDocument(1, "User", FileToUpload, string.Empty, this.DocumentForm.Description, "FAX", this.Language, this.DocumentTypes, string.Empty);
+                this.DocumentInfo = new DocumentInfo
+                {
+                    Cdn = this.Applicant.Cdn,
+                    DateStartDte = DateTime.UtcNow,
+                    DocumentId = result[0]
+                };
+
+
+                ClientXrefDocumentRepository.Insert(this.DocumentInfo);
+                this.State.FileUploadDTO.FileUploadComplete = true; //bug if navigated to this page directly
+                this.NavigationManager.NavigateTo($"/SeafarerProfile/{this.Applicant.Cdn}");
+            }
+        }
+       
+        
+        /// <summary>
+        /// Checks if the form is validated
+        /// </summary>
+        /// <returns></returns>
+        private bool Validate()
+        {
+            if (this.DocumentForm.SelectValue <= -1)
+            {
+               return false;
+
+            }
+           
+            if (this.DocumentTypes.Count <= 0)
+            {
+
+                return false;
+            }
+            if(this.FileToUpload == null)
+            {
+                return false;
+            }
+            return true;
+        }
+        /// <summary>
+        /// populates the list of document types from the form
+        /// </summary>
+        /// <param name="list"></param>
+        /// <returns></returns>
+        private List<string> PopulateDocumentTypes(List<SelectListItem> list)
+        {
+            List<string> DocumentTypes = new List<string>();
+            foreach (var i in list)
+            {
+                if (i.Value)
+                {
+                    DocumentTypes.Add(i.Text);
+                }
+            }
+            return DocumentTypes;
+        }
+        
+
+        /// <summary>
+        /// cancels and returns to the profile page
+        /// </summary>
+        public void HandleCancel()
+        {
+            if (this.FileToUpload != null)
+            {
+                this.FileToUpload = null;
+            }
+            this.NavigationManager.NavigateTo($"/SeafarerProfile/{this.Cdn}");
```

Comment:
We could simplify to this. 
```suggestion
            this.FileToUpload = null;
            this.NavigationManager.NavigateTo($"/SeafarerProfile/{this.Cdn}");
```